subsys/zigbee/cli/zigbee_cli_ctx_mgr.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <shell/shell.h>
+
+#include "zigbee_cli_ctx_mgr.h"
+
+
+/* The table to store context manager entries. */
+static struct ctx_entry zb_cli_ctx[CONFIG_ZIGBEE_SHELL_CTX_MGR_ENTRIES_NBR];
+
+/**@brief Delete context manager entry to mark it free and prepare to be
+ *        reused later.
+ *
+ * @param ctx_entry       Pointer to the context manager entry to be deleted.
+ */
+void ctx_mgr_delete_entry(struct ctx_entry *ctx_entry)
+{
+	memset(ctx_entry, 0, sizeof(struct ctx_entry));
+}
+
+/**@brief Get a pointer to a unused context manager entry.
+ *
+ * @param type            Type of a data to be stored in an entry.
+ *
+ * @return  Pointer to the context manager entry if an unused entry was found,
+ *          NULL otherwise.
+ */
+struct ctx_entry *ctx_mgr_new_entry(enum ctx_entry_type type)
```

Comment:
Will this function be used from different contexts? If so, I'd add `sys_mutex` protection here, curretnly it's not thread-safe.

Reply:
That's a good point, will add